glrcfg/src/runner/executors/docker.rs

```diff
@@ -10,35 +9,143 @@ macro_rules! stringvec {
 /// or any container runtime configured inside a job, does not inherit these parameters.
 ///
 /// Further documentation found in [the GitLab
-/// docs](https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-global-section).
+/// docs](https://docs.gitlab.com/runner/configuration/advanced-configuration.html#the-runnersdocker-section).
 #[derive(Debug, Serialize)]
 pub struct Docker {
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub allowed_images: Vec<String>,
+    pub allowed_privileged_images: Vec<String>,
+    pub allowed_pull_policies: Vec<String>,
+    pub allowed_services: Vec<String>,
+    pub allowed_privileged_services: Vec<String>,
+    pub cache_dir: Option<String>,
+    pub cap_add: Vec<String>,
+    pub cap_drop: Vec<String>,
+    pub cpuset_cpus: Option<String>,
+    pub cpuset_mems: Option<String>,
+    pub cpu_shares: u32,
+    pub cpus: Option<String>,
+    pub devices: Vec<String>,
+    pub device_cgroup_rules: Vec<String>, // https://docs.docker.com/compose/compose-file/05-services/#device_cgroup_rules
+    pub disable_cache: bool,              // written in cli runner creation
+    pub disable_entrypoint_overwrite: bool, // written in cli runner creation
+    pub dns: Vec<String>,
+    pub dns_search: Vec<String>,
+    pub extra_hosts: Vec<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub allowed_images: Option<String>,
-    pub disable_cache: bool,
-    pub disable_entrypoint_overwrite: bool,
-    pub image: String,
-    pub network_mtu: u32,
-    pub oom_kill_disable: bool,
-    pub privileged: bool,
-    pub smg_size: u32,
-    pub tls_verify: bool,
-    pub volumes: Vec<String>,
+    pub gpus: Option<String>,
+    pub group_add: Vec<String>,
+    pub helper_image: Option<String>,
+    pub helper_image_flavor: Option<String>,
+    pub helper_image_autoset_arch_and_os: Option<String>,
+    pub host: Option<String>,
+    pub hostname: Option<String>,
+    pub image: String, // written in cli runner creation
+    pub links: Vec<String>,
+    pub memory: Option<String>,
+    pub memory_swap: Option<String>,
+    pub memory_reservation: Option<String>,
+    pub network_mode: Option<String>,
+    pub network_mtu: u32, // written in cli runner creation, not in gitlab runner docs
+    pub mac_address: Option<String>,
+    pub oom_kill_disable: bool, // written in cli runner creation
+    pub oom_score_adjust: Option<i32>,
+    pub privileged: bool, // written in cli runner creation
+    pub pull_policy: Vec<String>,
```

Comment:
Input can actually be String as well as a list of strings.

The list allows for a chaining of multiple policies if the first one fails